cumulusci/tasks/apex/custom_settings.py

```diff
@@ -0,0 +1,101 @@
+""" a task for waiting on a specific custom settings value """
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.exceptions import SalesforceException
+from simple_salesforce import SalesforceMalformedRequest
+from cumulusci.core.utils import process_bool_arg
+
+
+class CustomSettingValueWait(BaseSalesforceApiTask):
+    """ CustomSettingValueWait polls an org until the specific value exists in a custom settings field """
+
+    name = "CustomSettingValueWait"
+
+    task_options = {
+        "object": {
+            "description": "Name of the Hierarchical Custom Settings object to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "field": {
+            "description": "Name of the field on the Custom Settings to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "value": {
+            "description": "Value of the field to wait for (String, Integer or Boolean). ",
+            "required": True,
+        },
+        "poll_interval": {
+            "description": (
+                "Seconds to wait before polling for batch job completion. "
+                "Defaults to 10 seconds."
+            ),
+        },
+    }
+
+    def _run_task(self):
+        self.poll_interval_s = int(self.options.get("poll_interval", 10))
+
+        self.object_name = self.options["object"]
+        self.field_name = self.options["field"]
+        self.check_value = self.options["value"]
+
+        # Process namespace tokens
+        namespace = self.project_config.project__package__namespace
```

Comment:
@prescod and @davisagli I had though (clearly incorrectly) that this would return a namespace if it was needed (kind of like what would happen in Apex). Is there a better way to do this so that the Namespace is applied when it's _needed_, or is the best approach to use the same `managed` and `namespace` option logic that the apex anonymous task implements?

Reply:
I ended up copying the logic from ApexAnonymous so this now uses the `managed` and `namespaced` options. Works much better now.